pkg/kubelet/prober/prober_manager_test.go

```diff
@@ -130,6 +182,20 @@ func TestAddRemovePods(t *testing.T) {
 	if err := expectProbes(m, nil); err != nil {
 		t.Error(err)
 	}
+
+	// Adding duplicate probes, should not add probe worker
+	m.AddPod(&dupReadinessProbePod)
+	if len(m.workers) != 1 {
+		t.Fatal("expected no duplicate readiness probe worker to be created")
+	}
+	m.AddPod(&dupStartupProbePod)
+	if len(m.workers) != 2 {
+		t.Fatal("expected no duplicate startup probe worker to be created")
+	}
+	m.AddPod(&dupLivenessProbePod)
+	if len(m.workers) != 3 {
+		t.Fatal("expected no duplicate liveness probe worker to be created")
+	}
```

Comment:
Would it make sense to modifying this test into a set of well-defined test cases?

Reply:
Polished the test cases in the new commit, PTAL